delegator.py

```diff
@@ -107,14 +108,15 @@ def return_code(self):
     def std_in(self):
         return self.subprocess.stdin
 
-    def run(self, block=True):
+    def run(self, block=True, binary_stdio=False):
         """Runs the given command, with or without pexpect functionality enabled."""
         self.blocking = block
 
         # Use subprocess.
         if self.blocking:
-            s = subprocess.Popen(self._popen_args, **self._default_popen_kwargs)
-
+            popen_kwargs = self._default_popen_kwargs
+            popen_kwargs['universal_newlines'] = not binary_stdio
```

Comment:
Ha, good catch about the `copy()`, as per `universal_newlines` it forces the output to be treated like a string when set to true (it was already set in the defaults).